File: JavaScript/arrays_and_hashing/removeDuplicates.js
var removeDuplicates = (nums) => {
    // set pointers to the initial value
    [left, right] = [0, 0];
    // move right pointer through every element
    while (right < nums.length) {
        const [leftVal, rightVal] = [nums[left], nums[right]];
        // check if current val is equal to previous val
        const isEqual = (leftVal === rightVal);
        if  (!isEqual) {
            // move left pointer to next index
            left++;
            nums[left] = rightVal;
        }
        right++;
    }
    return (left + 1);
};

File: arrays_&_hashing/remove_duplicates.py
from typing import List

class RemoveDuplicates:
    def remove_duplicates(self, nums: List) -> int:
        # initialize a pointer
        left = 1
        if nums == 0:
            return 0
        for i in range(1, len(nums)):
            # check if current value is equal to previous value
            if nums[i] != nums[i - 1]:
                nums[left] = nums[i]
                # increment left pointer
                left += 1
        return left